app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -35,11 +35,20 @@ def set_user_by_token(mapping = nil)
     access_token_name = DeviseTokenAuth.headers_names[:'access-token']
     client_name = DeviseTokenAuth.headers_names[:'client']
 
+    # gets values from cookie if configured and present
```

Comment:
can all this new cookie code be in a new module like `set_user_by_cookie` or something like that, so we don't pollute more this big file, you can include the concern/module in this file

Reply:
> I don't like using cookies in APIs but your reasoning of why is a good alternative sounds good

Yeah, we weren't too keen on it either. But it's the only way to share state across subdomains. And the general consensus seems to be that it's more secure for auth (once you're using **only** the HttpOnly cookie, which we'll hopefully allow in later work).

Reply:
I can, my only push back would be that hopefully in the future we'll allow turning off headers (for the security reasons mentioned in point two of the description). In that case, it might feel a bit disjointed to have the header logic in `set_user_by_token` while the cookie logic is in `set_user_by_cookie`, when really both logics deal with handling the token. In that future world, would we move the header logic to another concern called `set_user_by_headers`? If so, wouldn't `set_user_by_token` just become a skeleton file?

Reply:
On this note, see [my recent realization](https://github.com/lynndylanhurley/devise_token_auth/pull/1453#discussion_r553446626). That will clean up the amount of noise in this file considerably, commit incoming.